types/types.go

```diff
@@ -26,13 +26,13 @@ type AddDeployResponse struct {
 	ContractAddress string `json:"address"`
 }
 
-// TODO: remove
-type DeployRequest struct {
-	Type                string        `json:"type"`
-	ContractAddressSalt string        `json:"contract_address_salt"`
-	ConstructorCalldata []string      `json:"constructor_calldata"`
-	ContractDefinition  ContractClass `json:"contract_definition"`
-}
+// // TODO: remove
+// type DeployRequest struct {
+// 	Type                string               `json:"type"`
+// 	ContractAddressSalt string               `json:"contract_address_salt"`
+// 	ConstructorCalldata []string             `json:"constructor_calldata"`
+// 	ContractDefinition  rpcv02.ContractClass `json:"contract_definition"`
+// }
```

Comment:
I'm a bit confused about DeployRequest struct. I see that it's still being used in some tests and examples, so those shouldn't be working at the moment. I've seen in other files such as `gateway/starknet.go`, it's being used from `rpcv02` (it's commented atm), but it's not defined there. So is it the idea to move it to `rpcv02` types or to replace it with `rpcv02.DeployAccountTxn` or simply to remove it completely?

Reply:
The goal is to move away from using the (ambiguous) `types` package, and use more well-defined packages like `rpcv02`, `rpcv03` etc. 

The motivation for this is that (afaik) the `types` package does not follow any well-defined spec outlined by Starknet (plus the fields in the structs are incorrect). `RPCv02` and `RPCv03` do/will follow well-defined specs outlined by Starknet. I feel like this ambiguity will create unnecessary confusion and problems in the future, which we can prevent by deprecating `types` and using `rpcv02`. 

For example, if I'm a new user of starknet.go, and want to send a types.DeployAccountTxn to some Sequencer, which representation am I using? Does my request follow the rpcv02 spec, the rpcv03 spec, the gateway spec, or something else entirely? By deprecating the `types` package and using packages like `rpcv02`, `rpcv03`, `gateway`, we remove this ambiguity and prevent any problems that might arise because of it.

Reply:
I agree that this test breaks some already broken tests, but as we discussed previously we decide to leave fixing tests to other PRs. I'll have a look at fixing the examples now though.